azure/services/managedclusters/managedclusters.go

```diff
@@ -79,6 +79,8 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		return nil
 	}
 
+	// TODO: @nawazkh do you put the diff here?
+	// If yes, how do you compare the typeCasted struct b/w managedClusterSpec and containerservice.ManagedCluster ?
```

Comment:
@nawazkh we are already calculating the diff in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/managedclusters/spec.go#L330, the ask is to print it (and maybe also print a log message also when diff is nil) at high verbosity (to avoid making logs too noisy)

Reply:
the only issue is that we're not currently passing in a context for the Parameters() func so this also requires changing the func signature and thus the `ResourceSpecGetter` interface to pass in a context

Reply:
Thank you for the suggestion! I am able to puzzle the code flow together!
Testing with some local log changes before I make a change at interface level.